src/sonarqube.ts

```diff
@@ -1,14 +1,148 @@
+    'sonarqube.projectKeyTemplate'?: string
+}
+
+const getConfig = (): Configuration => sourcegraph.configuration.get<Configuration>().value
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    const config = getConfig()
+    const sonarqubeUrl = new URL(config['sonarqube.instanceUrl'] || 'https://sonarcloud.io/')
+    const corsAnyWhereUrl = new URL(config['sonarqube.corsAnywhereUrl'] || 'https://cors-anywhere.herokuapp.com')
+    const sonarqubeApiUrl = new URL(
+        `${corsAnyWhereUrl.href.replace(/\/$/, '')}/${sonarqubeUrl.href.replace(/\/$/, '')}/`
+    )
+    context.subscriptions.add(
+        combineLatest([
+            from(sourcegraph.app.activeWindowChanges).pipe(
+                switchMap(activeWindow => activeWindow?.activeViewComponentChanges || EMPTY),
+                filter((viewer): viewer is sourcegraph.CodeEditor => !!viewer && viewer.type === 'CodeEditor')
+            ),
+            from(sourcegraph.configuration).pipe(map(() => getConfig())),
+        ])
+            .pipe(
+                switchMap(async ([editor, config]) => {
+                    try {
+                        if (config['sonarqube.showIssuesOnCodeViews'] === false) {
+                            return { editor, issues: [] }
+                        }
+
+                        const uri = new URL(editor.document.uri)
+                        const repoName = decodeURIComponent(uri.hostname + uri.pathname)
+                        const commitID = decodeURIComponent(uri.search.slice(1))
+                        const filePath = decodeURIComponent(uri.hash.slice(1))
+
+                        const repositoryNamePattern = new RegExp(
+                            config['sonarqube.organizationPattern'] || '(?:^|/)([^/]+)/([^/]+)$'
+                        )
+                        const repositoryNameMatch = repoName.match(repositoryNamePattern)
+                        if (!repositoryNameMatch) {
+                            throw new Error(
+                                `repositoryNamePattern ${repositoryNamePattern.toString()} did not match repository name ${repoName}`
+                            )
+                        }
+                        const organizationKeyTemplate = config['sonarqube.organizationKeyTemplate'] ?? '$1'
+                        const organization = organizationKeyTemplate.replace(
+                            /\$(\d)/g,
+                            (substring, number: string) => repositoryNameMatch[+number]
+                        )
+                        const projectKeyTemplate = config['sonarqube.projectKeyTemplate'] ?? '$1_$2'
+                        const project = projectKeyTemplate.replace(
+                            /\$(\d)/g,
+                            (substring, number: string) => repositoryNameMatch[+number]
+                        )
+                        console.log('Mapped repository name to Sonarqube according to templates', {
```

Comment:
Should we keep this?

Reply:
I think it can be helpful for debugging patterns and templates